src/NuGet.Core/NuGet.Protocol.Core.v3/DownloadResourceV3.cs

```diff
@@ -116,10 +118,15 @@ private DownloadResourceV3(HttpClient client)
                 throw new ArgumentNullException(nameof(settings));
             }
 
+            if (logger == null)
+            {
+                logger = new NullLogger();
```

Comment:
Fixed in https://github.com/NuGet/NuGet.Client/commit/b3847c528179be0cc7e4e4e6e08e2448268c8af1